exec/Main.hs

```diff
@@ -31,44 +34,50 @@ import           Text.Printf
-          Left e -> printf "[FAIL] Unable to connect to %s /info endpoint\n%s" (showUrlScheme us) e >> exitFailure
-          Right ni -> do
-            let !mcids = map (second (map (ChainId . fst))) <$> _nodeInfo_graphs ni
-            case mcids of
-              Nothing -> printf "[FAIL] Node did not have graph information" >> exitFailure
-              Just cids -> do
-                let !env = Env m pool us u ni cids
-                case c of
-                  Listen -> listen env
-                  Backfill as -> backfill env as
-                  Gaps rateLimit -> gaps env rateLimit
-                  Single cid h -> single env cid h
-                  Server serverEnv -> apiServer env serverEnv
+  withHandleBackend backendConfig $ \backend ->
+    withLogger config backend $ \logger -> do
+      let logg = loggerFunIO logger
+      case args of
+        RichListArgs (NodeDbPath mfp) -> do
+          fp <- case mfp of
+            Nothing -> do
+              h <- getHomeDirectory
+              let h' = h </> ".local/share"
+              logg Info $ "[INFO] Constructing rich list using default db-path: " <> fromString h'
+              -- putStrLn $ "[INFO] Constructing rich list using default db-path: " <> h'
+              return h'
+            Just fp -> do
+              logg Info $ "[INFO] Constructing rich list using given db-path: " <> fromString fp
+              return fp
+          richList logg fp
+        Args c pgc us u -> do
+          logg Info $ "Using database: " <> fromString (show pgc)
+          logg Info $ "Service API: " <> fromString (showUrlScheme us)
+          logg Info $ "P2P API: " <> fromString (showUrlScheme (UrlScheme Https u))
+          withPool pgc $ \pool -> do
+            P.withResource pool initializeTables
+            logg Info "DB Tables Initialized"
+            let mgrSettings = mkManagerSettings (TLSSettingsSimple True False False) Nothing
+            m <- newManager mgrSettings
+            getNodeInfo m us >>= \case
+              Left e -> logg Error (fromString $ printf "[FAIL] Unable to connect to %s /info endpoint\n%s" (showUrlScheme us) e) >> exitFailure
+              Right ni -> do
+                let !mcids = map (second (map (ChainId . fst))) <$> _nodeInfo_graphs ni
+                case mcids of
+                  Nothing -> logg Error "[FAIL] Node did not have graph information" >> exitFailure
+                  Just cids -> do
+                    let !env = Env m pool us u ni cids logg
+                    case c of
+                      Listen -> listen env
+                      Backfill as -> backfill env as
+                      Gaps rateLimit -> gaps env rateLimit
+                      Single cid h -> single env cid h
+                      Server serverEnv -> apiServer env serverEnv
   where
     opts = info ((richListP <|> envP) <**> helper)
       (fullDesc <> header "chainweb-data - Processing and analysis of Chainweb data")
+    config = defaultLoggerConfig
```

Comment:
Let's make a command line argument `--level` that the user can use to change the `_loggerConfigThreshold` field.